llvm/test/Feature/Repo/repo_discardable_if_unused_test.ll

```diff
@@ -0,0 +1,22 @@
+; Check the caller's pruning  if its callee has noinline attribute and may be discarded if the callee is not used.
```

Comment:
I think that the description in the PR and the bug are clearer than this and would be useful to incorporate into the comment here and/or in RepoTicket.cpp.

Quote from PR:
> If the caller and callee satisfies the following conditions, the error of undefined reference to the callee is given.
> 
> Caller is pruned and deleted during the RepoPruning pass.
> Callee isn’t pruned, has noinline attribute and may be discarded if it is not used.
> The reason is that the callee is removed during the Dead Global Elimination pass, which is after the repo pruning pass. The current hash calculation algorithm doesn’t accumulate the callee's hash value to the caller if the callee has noinline attribute.

Quote from bug:
> When we modify the variable e value, the hash value of f is changed. However, the hash value of its caller g keeps the same. During the second time compile, the function g is pruned and f is not pruned. Since the function g is pruned and has an empty UseList, it will be deleted. After repo pruning, the function g has been changed from definition to declaration, which will result in changing the UseList of function f from 1 to 0.
> 
> The Dead Global Elimination pass is after the repo pruning pass. Since the function f is local function and has an empty UseList, it is removed as well. Therefore, there is only one compilation member g for the file foo.c during the second build. There are two missing compilation members: function f and variable f.e. This causes the Link error.

Reply:
All comments are added into the test, thanks.